kratos/utilities/data_type_traits.h

```diff
@@ -67,6 +67,20 @@ template<class TDataType> class DataTypeTraits
         return IsDimensionDynamicImpl<TCheckIndex, 0>();
     }
 
+    /**
+     * @brief Checks if the given shape is valid.
+     *
+     * @param pShapeBegin           Begining of the shape array.
+     * @param pShapeEnd             End of the shape array.
+     */
+    template<class TIteratorType>
+    static bool inline IsValidShape(
+        TIteratorType pShapeBegin,
+        TIteratorType pShapeEnd)
```

Comment:
1) The shape array is supposed to be contiguous, right? If so, you don't need to template on the iterator type, but the index type and use pointers to it as your iterator.

2) `inline` does nothing here.

```suggestion
    template<class TInteger>
    static bool inline IsValidShape(
        const TInteger* pShapeBegin,
        const TInteger* pShapeEnd)
```

Reply:
I am ok, then I always have to take the underlying containers of the shape, (sometimes it is `std::vector`, sometimes it is `DenseVvector`)

But I agree. I will change. Just wanted to point this out

Reply:
done :)